examples/java/run.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -euo pipefail
```

Comment:
file diff doesn't show, this was the old version:

```bash
#!/bin/bash

cd example-app
if [[ ! -f ./target/example-app.jar ]] ; then
    ./mvnw clean package
fi
if [[ ! -f ./opentelemetry-javaagent.jar ]] ; then
    curl -sOL https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v2.0.0/opentelemetry-javaagent.jar
fi
export OTEL_EXPORTER_OTLP_PROTOCOL=grpc
export OTEL_SEMCONV_STABILITY_OPT_IN=http
export OTEL_RESOURCE_ATTRIBUTES="service.name=example-app,service.instance.id=localhost:8080"
java -Dotel.metric.export.interval=500 -Dotel.bsp.schedule.delay=500 -javaagent:opentelemetry-javaagent.jar -jar ./target/example-app.jar
```

Reply:
in the new version

- removed `export OTEL_SEMCONV_STABILITY_OPT_IN=http` that is no longer needed
- don't reuse old downloaded jar if the version doesn't match